test/jdk/java/awt/print/PrinterJob/raster/RasterTest.java

```diff
@@ -21,248 +21,134 @@
+
+    public static void main(String[] args) throws Exception {
+
+        if (PrinterJob.lookupPrintServices().length == 0) {
+            throw new RuntimeException("Printer not configured or available.");
+        }
+
+        PassFailJFrame.builder()
+                .instructions(INSTRUCTIONS)
+                .testUI(RasterTest::new)
+                .rows((int) INSTRUCTIONS.lines().count() + 1)
+                .columns(45)
+                .build()
+                .awaitAndCheck();
+    }
 
- private RasterCanvas c;
-
- public static void main(String args[]) {
-  String[] instructions =
-        {
-         "You must have a printer available to perform this test",
-         "This test uses rendering operations which force the implementation",
-         "to print the page as a raster",
-         "You should see two square images, the 1st containing overlapping",
-         "composited squares, the lower image shows a gradient paint.",
-         "The printed output should match the on-screen display, although",
-         "only colour printers will be able to accurately reproduce the",
-         "subtle color changes."
-       };
-      Sysout.createDialog( );
-      Sysout.printInstructions( instructions );
-
-    RasterTest f = new RasterTest();
-    f.show();
- }
-
- public RasterTest() {
+    public RasterTest() {
         super("Java 2D Raster Printing");
 
-    c = new RasterCanvas();
-    add("Center", c);
-
-    Button printButton = new Button("Print");
-    printButton.addActionListener(this);
-    add("South", printButton);
-
-    addWindowListener(new WindowAdapter() {
-       public void windowClosing(WindowEvent e) {
-             System.exit(0);
-            }
-    });
-
-    pack();
+        c = new RasterCanvas();
+        add("Center", c);
 
+        Button printButton = new Button("Print");
+        printButton.addActionListener(this);
+        add("South", printButton);
+        pack();
         setBackground(Color.white);
-
- }
-
- public void actionPerformed(ActionEvent e) {
-
-   PrinterJob pj = PrinterJob.getPrinterJob();
-
-   if (pj != null && pj.printDialog()) {
-       pj.setPrintable(c);
-       try {
-            pj.print();
-      } catch (PrinterException pe) {
-      } finally {
-         System.err.println("PRINT RETURNED");
-      }
-   }
-}
-
-
- class RasterCanvas extends Canvas implements Printable {
-
-
-    public int print(Graphics g, PageFormat pgFmt, int pgIndex) {
-      if (pgIndex > 0)
-         return Printable.NO_SUCH_PAGE;
-
-         Graphics2D g2d= (Graphics2D)g;
-         g2d.translate(pgFmt.getImageableX(), pgFmt.getImageableY());
-         doPaint(g2d);
-      return Printable.PAGE_EXISTS;
     }
 
-    public void paint(Graphics g) {
-       doPaint(g);
-    }
+    public void actionPerformed(ActionEvent e) {
+        PrinterJob pj = PrinterJob.getPrinterJob();
 
-    public void paintComponent(Graphics g) {
-       doPaint(g);
+        if (pj != null && pj.printDialog()) {
```

Comment:
```suggestion
        if (pj.printDialog()) {
```
The contract of [`PrinterJob.getPrinterJob`](https://docs.oracle.com/en/java/javase/21/docs/api/java.desktop/java/awt/print/PrinterJob.html#getPrinterJob()) doesn't allow returning `null`, so the null-check can be removed, which will resolve a warning from IDE.